vslib/src/MACsecFilter.cpp

```diff
@@ -30,6 +32,13 @@ void MACsecFilter::enable_macsec_device(
     SWSS_LOG_ENTER();
 
     m_macsecDeviceEnable = enable;
+    // The function, MACsecFilter::execute(), may be running in another thread,
+    // the macsec device enable state cannot be changed in the busy state.
+    // Because if the macsec device was deleted in the busy state,
+    // the error value of function, MACsecFilter::forward, will be returned
+    // to the caller of MACsecFilter::execute()
+    // and the caller thread will exit due to the error return value.
+    while(get_state() != MACsecFilter::MACsecFilterState::IDLE);
```

Comment:
dont put ; after while statment in the same line, please make this proper { } and insert comment // loop etc